src/gazebo/differential_drive_6w.cpp

```diff
@@ -62,130 +64,122 @@ namespace gazebo {
+            mJoints[REAR_RIGHT] = mModel->GetJoint(mSdf->GetElement("rearRightJoint")->GetValue()->GetAsString());
+
+        if (!mJoints[FRONT_LEFT]) ROS_FATAL("diffdrive_plugin_6w error: The controller couldn't get front left joint");
+        if (!mJoints[FRONT_RIGHT]) ROS_FATAL("diffdrive_plugin_6w error: The controller couldn't get front right joint");
+        if (!mJoints[MID_LEFT]) ROS_FATAL("diffdrive_plugin_6w error: The controller couldn't get mid left joint");
+        if (!mJoints[MID_RIGHT]) ROS_FATAL("diffdrive_plugin_6w error: The controller couldn't get mid right joint");
+        if (!mJoints[REAR_LEFT]) ROS_FATAL("diffdrive_plugin_6w error: The controller couldn't get rear left joint");
+        if (!mJoints[REAR_RIGHT]) ROS_FATAL("diffdrive_plugin_6w error: The controller couldn't get rear right joint");
+
+        if (mSdf->HasElement("wheelSeparation"))
+            mSdf->GetElement("wheelSeparation")->GetValue()->Get(mWheelSeparation);
+
+        if (mSdf->HasElement("wheelDiameter"))
+            mSdf->GetElement("wheelDiameter")->GetValue()->Get(mWheelDiameter);
+
+        if (mSdf->HasElement("torque"))
+            mSdf->GetElement("torque")->GetValue()->Get(mTorque);
 
         // Make sure the ROS node for Gazebo has already been initialized
         if (!ros::isInitialized()) {
-            ROS_FATAL_STREAM("A ROS node for Gazebo has not been initialized, unable to load plugin. "
-                             << "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the gazebo_ros package)");
+            ROS_FATAL("A ROS node for Gazebo has not been initialized, unable to load plugin. "
+                      "Load the Gazebo system plugin 'libgazebo_ros_api_plugin.so' in the gazebo_ros package)");
             return;
         }
 
-        rosnode_ = new ros::NodeHandle(namespace_);
+        mNode = ros::NodeHandle(mNamespace);
 
-        tf_prefix_ = tf::getPrefixParam(*rosnode_);
-        transform_broadcaster_ = new tf::TransformBroadcaster();
+        mTfPrefix = tf::getPrefixParam(*mNode);
+        mTfBroadcaster = tf::TransformBroadcaster();
 
         // ROS: Subscribe to the velocity command topic (usually "cmd_vel")
         ros::SubscribeOptions so = ros::SubscribeOptions::create<geometry_msgs::Twist>(
-                topic_, 1,
-                boost::bind(&DiffDrivePlugin6W::cmdVelCallback, this, _1),
-                ros::VoidPtr(), &queue_);
-        sub_ = rosnode_->subscribe(so);
-        pub_ = rosnode_->advertise<nav_msgs::Odometry>("odom", 1);
+                mVelocityCommandTopic, 1,
+                [this](const geometry_msgs::Twist::ConstPtr& velocityCommand) { commandVelocityCallback(velocityCommand); },
+                ros::VoidPtr(), &mVelocityCommandQueue);
+        mSubscriber = mNode->subscribe(so);
+        mPublisher = mNode->advertise<nav_msgs::Odometry>("odom", 1);
+        mSpinner = ros::AsyncSpinner(1, &mVelocityCommandQueue); // 1 core for now
```

Comment:
I pray that this doesn't actually pin a core.

Reply:
Hopefully it is better designed than that.